backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/domain/model/ItemStock.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2023 Volkswagen AG
+ * Copyright (c) 2023 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e.V.
+ * (represented by Fraunhofer ISST)
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.puris.backend.stock.domain.model;
+
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.EmbeddedId;
+import jakarta.persistence.Entity;
+import lombok.*;
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm.DirectionCharacteristic;
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm.ItemUnitEnumeration;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Objects;
+
+@NoArgsConstructor
+@Entity
+@ToString
+public class ItemStock {
+
+    @Getter
+    @EmbeddedId
+    private Key key = new Key();
+    private Date lastUpdatedOnDateTime;
```

Comment:
nullable and pattern evaluation I would like to have as annotation instead of a service validation. Ensures that the database can't be corrupted.

Reply:
See above